src/api.rs

```diff
@@ -82,18 +82,26 @@ async fn routes(state: AppState) -> Router {
 ///
 /// * `event_mappings` - A vector of `IndexerContractMapping` representing the event mappings.
 /// * `connection_string` - The connection string for the database.
-pub async fn start_api(event_mappings: &[IndexerContractMapping], connection_string: &str) {
-    let routes = routes(AppState {
-        connection_string: connection_string.to_string(),
-        event_mappings: event_mappings.to_vec(),
-    })
-    .await;
-
-    println!("reth indexer API started, listening on localhost:3030");
-    axum::Server::bind(&SocketAddr::from(([127, 0, 0, 1], 3030)))
-        .serve(routes.into_make_service())
-        .await
-        .expect("reth indexer API failed");
+pub async fn start_api(
+    event_mappings: &[IndexerContractMapping],
+    indexer_postgres_config: &Option<IndexerPostgresConfig>,
+) {
+    // TODO: api should be able to accommodate different sources, but currently only for postgres
```

Comment:
can we throw a message if you try to run this connected to a bigquery config so they can see it in the command when running